drivers/ethernet/eth_stm32_hal.c

```diff
@@ -806,6 +804,12 @@ static void eth_iface_init(struct net_if *iface)
 	ethernet_init(iface);
 
 	net_if_flag_set(iface, NET_IF_NO_AUTO_START);
+
+	if (is_first_init == true) {
+		/* Now that the iface is setup, we are safe to enable IRQs. */
+		__ASSERT_NO_MSG(DEV_CFG(dev)->config_func != NULL);
+		DEV_CFG(dev)->config_func();
```

Comment:
```suggestion
		DEV_CFG(dev)->config_func();
    is_first_init = false;
```

Do you want maybe reset the flag then?

Reply:
No need here, the flag is initialised to **is_first_init="false"** at the start of the function. 😄